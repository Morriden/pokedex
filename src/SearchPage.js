import React, { Component } from 'react'
import request from 'superagent'
import './App.css'
import PokemonUL from './PokemonUL.js'
import SearchSection from './SearchSection.js'

export default class SearchPage extends Component {


state = {
  search: '',
  dropdown: 'pokemon',
  data: [],
}

async componentDidMount() {
  const data = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex`)
  this.setState({ data: data.body.results })
}
  
handleDropDownChange = (e) => {
  const value = e.target.value;
  this.setState({ dropdown: value});
  
}

  handleChange = (e) => {
    
    const value = e.target.value;
    this.setState({ search: value});
  }

  handleClick = async () => {
    const search = this.state.search;
    this.setState({ loading: true })
    const fetchedData = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.dropdown}=${search}`);
    console.log(fetchedData)
    this.setState({ loading: false })
    this.setState({ data: fetchedData.body.results })
  }

  render() {
    console.log(this.state.data)

    return (
      <body> 
        <header>Pokedex</header>   
          <main>
              <SearchSection
            CallBackhandleChange={this.handleChange}
            CallBackhandleDropDownChange={this.handleDropDownChange}
            CallBackhandleClick={this.handleClick}
          /> 
          <div className="pokemon-area">
                <PokemonUL pokemon={this.state.data} />
          </div>
        </main>
      </body>
    )
  }
}  
         
        

