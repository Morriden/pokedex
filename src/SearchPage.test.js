import React from 'react';
import SearchPage from './App.js';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';

test('will come back without stuff', () => {
    const wrapper = shallow(<SearchPage />)

   expect(wrapper).toMatchSnapshot();

});

test('will come back with stuff', () => {
    const wrapper = shallow(<SearchPage name="some name"/>)

   expect(wrapper).toMatchSnapshot();

});