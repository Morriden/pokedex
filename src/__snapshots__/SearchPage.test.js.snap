// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will come back with stuff 1`] = `
<div>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        path="/pokemon/:pokemon"
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;

exports[`will come back without stuff 1`] = `
<div>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        path="/pokemon/:pokemon"
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
